src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -169,6 +192,58 @@ def pex_requirements_for_default_lockfile(cls):
             resolve_name=cls.options_scope,
         )
 
+    @classproperty
```

Comment:
Is there some equivalent of `functools.cached_property` for class properties? I imagine that parsing the lockfile every single time this is called is not great and the default lockfiles change infrequently enough that this can definitely be cached.

Reply:
Good thinking.  I suspect adding `functools.cache` decorator first does the right thing? (In particular, it'll be caching based on the value of the `cls` parameter, which seems appropriate.)

```python
from functools import cache

...

    @classproperty
    @cache
    def default_package_name_and_version(cls) -> Optional[PackageNameAndVersion]: 
```

Potentially we could reduce the magic by having this be a `@classmethod`, in which case `functools.cache` becomes more "obviously" correct?

Reply:
Changed to `classmethod` and added the cache, which seems to work well!